api/v1beta1/clusterclass_types.go

```diff
@@ -532,6 +532,12 @@ type ExternalPatchDefinition struct {
 	// ValidateExtension references an extension which is called to validate the topology.
 	// +optional
 	ValidateExtension *string `json:"validateExtension,omitempty"`
+
+	// Settings defines key value pairs to be passed to mutation hook.
+	// Value defined here take precedence over the values defined in the
+	// corresponding ExtensionConfig.
+	// +optional
+	Settings map[string]string `json:"settings,omitempty"`
```

Comment:
@ykakarap Can you please look into this?

> Incompatible changes:
>  - ExternalPatchDefinition: old is comparable, new is not
> https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/7562/pull-cluster-api-apidiff-main/1595553240982753280

Not sure if it is relevant

Reply:
This is expected. Maps are not comparable in go. Since a map (Settings) was added to `ExternalPatchDefinition` the entire struct now become un-comparable. This should be fine as we don't really ever do operations like `externalPatchDefinition1 == externalPatchDefinition2`.